awx/main/isolated/manager.py

```diff
@@ -187,6 +188,13 @@ def check(self, interval=None):
                                                       self.private_data_dir,
                                                       extravars=extravars)
             status, rc = runner_obj.status, runner_obj.rc
+            if self.check_callback is not None:
+                config_path = self.path_to('artifacts', self.ident, 'config')
+                # If the configuration artifact has been synced back, update the model
+                if os.path.exists(config_path):
+                    with open(config_path, 'r') as f:
+                        data = json.load(f)
+                    self.check_callback(data)
```

Comment:
Are we going to end up calling this multiple times?

Reply:
The way I'm doing it now, probably.  I suppose the only path out other than completing the loop is if an exception is raised, so it can come out of the loop without much, if any, loss.

Reply:
I've brought it out of the loop.